include/swift/Reflection/ReflectionContext.h

```diff
@@ -474,8 +487,29 @@ class ReflectionContext
     savedBuffers.push_back(std::move(Buf));
     return true;
   }
-         
-  bool readELF(RemoteAddress ImageStart) {
+
+  /// Parses metadata information from an ELF image. Because the Section
+  /// Header Table is stripped out from the running process, this method also
+  /// receives a buffer with the contents of the image's file, from
+  /// where it will read the missing information.
+  ///
+  ///
+  /// \param[in] ImageStart
+  ///     A remote address pointing to the start of the image in the running
+  ///     process.
+  ///
+  /// \param[in] FileBuffer
+  ///     A buffer which contains the contents of the image's file
+  ///     in disk.
+  ///
+  /// \param[in] FileSize
+  ///     The size of \p FileBuffer.
```

Comment:
Very nice! Can you document what's expected to happen if the FileBuffer is empty? (I.e., will it work, but with a degraded experience, or will it always return false?)